spec/presenters/tree_builder_belongs_to_hac_spec.rb

```diff
@@ -0,0 +1,152 @@
+  let(:group) { FactoryGirl.create(:miq_group)}
+  let(:datacenter1) do
+    datacenter = FactoryGirl.create(:datacenter)
+    datacenter.with_relationship_type("ems_metadata") { datacenter.add_folder(subfolder) }
+    datacenter
+  end
+  let(:datacenter2) { FactoryGirl.create(:datacenter) }
+  let(:rp1) do
+    rp1 = FactoryGirl.create(:resource_pool, :is_default => true)
+    rp1.with_relationship_type("ems_metadata") { rp1.add_child(rp2) }
+    rp1
+  end
+  let(:rp2) { FactoryGirl.create(:resource_pool) }
+  let(:cluster) do
+    cluster = FactoryGirl.create(:ems_cluster)
+    allow(cluster).to receive(:resource_pools) { [rp1] }
+    allow(cluster).to receive(:hosts) { [host] }
+    cluster
+  end
+  let(:ems_folder) { FactoryGirl.create(:ems_folder) }
+  let(:subfolder) do
+    subfolder = FactoryGirl.create(:ems_folder, :name => 'host')
+    subfolder.with_relationship_type("ems_metadata") { subfolder.add_child(datacenter2) }
+    subfolder.with_relationship_type("ems_metadata") { subfolder.add_child(ems_folder) }
+    subfolder.with_relationship_type("ems_metadata") { subfolder.add_host(host) }
+    subfolder.with_relationship_type("ems_metadata") { subfolder.add_cluster(cluster) }
+    subfolder
+  end
+  let(:subfolder_vm) do
+    subfolder_vm = FactoryGirl.create(:ems_folder, :name => 'vm')
+    subfolder_vm.with_relationship_type("ems_metadata") { subfolder_vm.add_child(ems_folder) }
+    subfolder_vm
+  end
+  let(:folder) do
+    folder = FactoryGirl.create(:ems_folder)
+    folder.with_relationship_type("ems_metadata") { folder.add_child(subfolder) }
+    folder.with_relationship_type("ems_metadata") { folder.add_child(datacenter1) }
+    folder
+  end
+  let(:host) do
+    FactoryGirl.create(:host, :ems_cluster => cluster)
+  end
+  let(:ems_azure_network) do
+    network = FactoryGirl.create(:ems_azure_network)
+    network.with_relationship_type("ems_metadata") { network.add_child(folder) }
+    network
+  end
+
+  subject do
+    described_class.new(:hac,
+                        :hac_tree,
+                        {:trees => {}},
+                        true,
+                        :edit     => edit,
+                        :filters  => {},
+                        :group    => group,
+                        :selected => {})
+  end
+
+
+
+  describe '#tree_init_options' do
+    it 'sets init options correctly' do
+      expect(subject.send(:tree_init_options, :hac)).to eq(:full_ids             => true,
+                                                           :add_root             => false,
+                                                           :lazy                 => false,
+                                                           :checkable_checkboxes => edit.present?,
+                                                           :selected             => {})
+    end
+  end
+
+  describe '#set_locals_for_render' do
+    it 'sets locals correctly' do
+      locals = subject.send(:set_locals_for_render)
```

Comment:
Could be these `expect`s underneath merged into one and compared with a hash? Then you would not need the `locals` variable...